src/MoBi.Core/Domain/Model/MoBiSpatialStructure.cs

```diff
@@ -1,19 +1,54 @@
-﻿using OSPSuite.Core.Diagram;
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Diagram;
 using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
 using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
 
 namespace MoBi.Core.Domain.Model
 {
    public interface IMoBiSpatialStructure : ISpatialStructure, IWithDiagramFor<IMoBiSpatialStructure>
    {
+      /// <summary>
+      ///    Return the list of neighborhood connected to this neighbor;
+      /// </summary>
+      IReadOnlyList<NeighborhoodBuilder> GetConnectingNeighborhoods(IReadOnlyList<IContainer> neighbors, IObjectPathFactory objectPathFactory);
    }
 
    public class MoBiSpatialStructure : SpatialStructure, IMoBiSpatialStructure
    {
       public IDiagramModel DiagramModel { get; set; }
       public IDiagramManager<IMoBiSpatialStructure> DiagramManager { get; set; }
 
+      public IReadOnlyList<NeighborhoodBuilder> GetConnectingNeighborhoods(IReadOnlyList<IContainer> neighbors, IObjectPathFactory objectPathFactory)
```

Comment:
THis code was duplicated. It's probably wrong with the new concept but I did not want to have to update it all the time